### PR TITLE
make unitree go2 mjx model compatible with hfields

### PR DESCRIPTION
**Pull request by 7oponaut**
I had two issues when attempting to use `unitree_go2/go2_mjx.xml` in scenes with hfields such as `google_barkour_vb/scene_hfield_mjx.xml`:

1) the `margin` attribute in the `go2` class default section results in the following error:

> NotImplementedError: (mjtGeom.mjGEOM_HFIELD, mjtGeom.mjGEOM_SPHERE) margin/gap not implemented.

2) the impratio="100" option setting makes the scene unstable with mjx acceleration (specifically, I encountered nan qpos values)

This PR removes these two attributes, resulting in no `margin` and an `impratio="1"` default.

I don't have a full understanding of how these changes affect the simulation. Please determine if they should be merged.

@saran-t requested this PR after encountering my post about the instability on X.

p.s. My version of the file has additional changes which might be helpful

- The go2-floor contact is way too slippery imo, it looks unrealistic to me. I 5x'ed friction on the feet of the go2 model, since they have higher priority than the floor.
- The tracking camera is too high for me. I set its z coordinate to 0.5 to center the go2 better.

image

### CODE REVIEW

**Comment by kevinzakka** [comment]
Thanks for these @7oponaut. We should probably check if the same changes should be made to the other Unitree MJX variants. Will wait on @erikfrey to review!

**Review by erikfrey — APPROVED** [review]
Both of these changes look good.  Although you can use impratio for pyramidal friction, it's more stable for elliptic cone friction.

The reason you are seeing more slippage and had to raise the friction is probably due to removing impratio, but that's OK.  I'll leave it to you and @kevinzakka to discuss whether to update friction, as he knows better than me where those values came from.

**Comment by kevinzakka** [comment]
@erikfrey I'm okay with increasing friction, but since elliptic is now supported by MJX, should we not switch to impratio + elliptic?

**Comment by erikfrey** [comment]
As a rule of thumb, for locomotion you can rely on pyramidal friction - it's cheaper to simulate but still transfers to real just fine.

**Comment by kevinzakka** [comment]
Gotcha, then all good for increasing friction!

**Comment by kevinzakka** [comment]
@erikfrey Can we get this merged in?

**Comment by kevinzakka** [comment]
@7oponaut Can you sign the CLA?

**Comment by erikfrey** [comment]
@kevinzakka yes, once CLA is signed.

**Comment by 7oponaut** [comment]
Sorry for taking your time, I was under the impression that I am contributing to an open source repository.

I am not comfortable with signing the CLA.

I'm taking issue with 3) in the CLA: I'm not a lawyer or a Google employee and I don't want to deal with litigatory issues in case Google later commercializes any of this repository and gets sued for whatever reason.

**Pull request closed without merging**
